rocky/reports/report_types/name_server_report/report.py

```diff
@@ -125,3 +126,46 @@ def generate_data(self, input_ooi: str, valid_time: datetime) -> dict[str, Any]:
             "name_server_checks": name_server_checks,
             "finding_types": sorted(finding_types.values(), reverse=True, key=lambda x: x.risk_severity),
         }
+
+    def collect_data(self, input_oois: Iterable[str], valid_time: datetime) -> dict[str, dict[str, Any]]:
+        hostnames_by_input_ooi = self.to_hostnames(input_oois, valid_time)
+        all_hostnames = [h for key, hostnames in hostnames_by_input_ooi.items() for h in hostnames]
+
+        query = "Hostname.<ooi[is Finding].finding_type"
+        hostname_finding_types = self.group_by_source(
+            self.octopoes_api_connector.query_many(query, valid_time, all_hostnames),
+            lambda ooi: ooi.id in ["KAT-NO-DNSSEC", "KAT-INVALID-DNSSEC"],
+        )
+        query = "Hostname.<hostname[is ResolvedHostname].address.<address[is IPPort].<ooi[is Finding].finding_type"
+        port_finding_types = self.group_by_source(
+            self.octopoes_api_connector.query_many(query, valid_time, all_hostnames),
+            lambda ooi: ooi.id in ["KAT-UNCOMMON-OPEN-PORT", "KAT-OPEN-SYSADMIN-PORT", "KAT-OPEN-DATABASE-PORT"],
```

Comment:
Since you're using the `check` argument quite often, this leads to a lot of repetitive code and lambda noise. I want to suggest to introduce a (partial) function that accepts a list of finding type names, something like:
```python3
def finding_type_checker(finding_types: list[str]):
    def _check(ooi, types):
        return ooi.id in types
     return functors.partial(_check, types=finding_types)

...
self.group_by_source(..., finding_types_checker(["KAT-UNCOMMON-OPEN-PORT", "...", ...]))
...
```